### DISCUSSION TITLE
Is there a better way to find a plot which win !

### DISCUSSION THREAD

**Discussion by pangchia** [Chia-Network/chia-blockchain#7796] [General]
after a block win , I find out the plot public key  [ 0x919...]

But it is horrible work to find the plot  by looking in chia client with 4 ~5k plots in  [ plot tab]。

it must has a better way find it !

Genius help me ！

**Comment by gryan315** [comment]
Why do you want to find it?

**Reply by rmica** [reply]
I may be wrong. Once I've read that there are security concerns on farming sizes >= 1 PB, suggesting to replot the winner. If this is true, it would be very helpful to easily know what file found the proof.

**Reply by gryan315** [reply]
Can you explain this? It does not sound like a logical concern to me?

**Reply by rmica** [reply]
I found a comment on https://chiaforum.com/t/are-winning-harvested-plots-removed/629 , quote: "Plots are good forever, they can continue to win indefinitely. Just because one wins doesn’t mean it is “used up”.
(There’s no reason to delete a winning plot, except if you are a mega-whale and have petabytes on the network there is some theoretical attack people can theoretically use against whales with winning plot info, so they advise deleting winning plots in that specific case only.)".
I can't explain it better because I don't know what type of theoretical attack he's talking about, just reporting. If, and I repeat, if, this is an issue, then to easily know the file would be a good service.

**Reply by gryan315** [reply]
A vague mention of a "theoretical" attack with no source and a reference to "they" (who is they?) advising people with petabytes of storage to delete winning plots? Sounds ridiculous to me. Seems like baseless speculation to me, I see no possible attack vector from an individual plot.

**Reply by rmica** [reply]
Ok, it's better this way. Thanks.

**Reply by rmica** [reply]
Looks like "they" is chia-network ...

**Reply by SlugFiller** [reply]
I'm doing some extrapolation from the available info, but basically, there is an attack that tricks a winning farmer into "re-winning" a block he's already won, by convincing him that the chain rolled back to just before that block. The good news is that it doesn't hurt the farmer, because they just end up winning the same block. The bad news is that it potentially hurts the rest of the network, because any transactions in that block are reverted, and replaced with a new set of transactions.

The reason for deleting the plot is to ensure a farmer can't "re-win" the block. This is good for the network, but is not good for the farmer, and in some cases, can even be bad for the farmer. If the chain re-orgs for any reason, whether it's an attack, or just a higher difficulty proof found, if the reorg hasn't changed the challenge, the farmer basically has the choice between re-farming a block, or leaving it up to other farmers so that the re-orged chain has a "fair competition" with the original one. If the re-orged chain wins anyway (due to another farmer successfully winning), the farmer that chose not to re-farm basically just lost.

In other words, deleting an old plot, to prevent a re-farm, is for the good of the network, NOT for the good of the farmer.

**Comment by keliew** [comment]
https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#i-have-heard-that-its-recommended-that-a-winning-plot-be-deleted-on-mainnet

**Reply by rmica** [reply]
So the info I had received was not "vague"? How to easily know the file? Thank you.

**Reply by keliew** [reply]
No easy way...and you knew it: https://github.com/Chia-Network/chia-blockchain/discussions/7748

**Reply by rmica** [reply]
True. I personally modified my source to log the file name with active proofs in the harvester. But it should be in the farmer's code, because in the harvester I can't separate NFT partials and blockchain proofs. I mean: I'm not into the source code at this level, just capable of adding a log.info call. I think that it would be a good service to write the winner file in a proper manner, probably by the farmer process, in order to enable scripts to eventually replot.

**Reply by rmica** [reply]
For people interested I edited  "chia-blockchain/chia/harvester/harvester_api.py".

At line 223 (version 1.2.3)

modified:
```
for response in sublist:
                total_proofs_found += 1
                msg = make_msg(ProtocolMessageTypes.new_proof_of_space, response)
                await peer.send_message(msg)
```
into:
```
for response in sublist:
                self.harvester.log.info(f"Proof on {filename}")
                total_proofs_found += 1
                msg = make_msg(ProtocolMessageTypes.new_proof_of_space, response)
                await peer.send_message(msg)
```
"sh install.sh" and restart harvester.

When proofs are >= 1, at log_level INFO, the message(s) will appear in the harvester's log before the classic "N plots were eligible for farming ...".

Far from perfect: NFT partials will write the info too.
Anyhow helpful, in my case.

**Reply by keliew** [reply]
But that'd be misleading...because https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#is-it-possible-to-have-a-proof-but-not-get-a-reward

But sure, you get closer to what you are looking for.

**Reply by pangchia** [reply]
it's helpful , thx
